안녕하세요.<br><br>
수강생 여러분~! Chapter05 ~ 08 수업은 잘 들으셨나요?

Chapter05 ~ 08의 수업 내용을 바탕으로 아래 문제를 풀어주시면 됩니다.

<br>


[**답변 입력 방법**]

문제 아래에 있는 코드 입력창에 정답코드를 작성하신 후  __▷ 버튼을 클릭해 코드를 실행__ 해보세요<br>

-----

__[문제0]__
<br>

##### 이전 과제에서 "도전! 신비의 섬 탐험" 을 변수를 이용해서 다시 풀어보세요 !!!!!



## 도전! 신비의 섬 탐험

![image.png](https://media.cnn.com/api/v1/images/stellar/prod/180219103122-zanzibar-and-its-islands-mnemba-a-view-from-the-sky-mnemba-island-lodge.jpg?q=w_1600,h_900,x_0,y_0,c_fill/h_447)

### [배경 이야기]

탐험가인 당신은 신비의 섬에 도착했습니다.


이 섬에는 숨겨진 보물이 있으며, 이를 찾기 위해 여러 도전 과제를 해결해야 합니다.


섬에는 알 수 없는 외계인이 당신을 쫓아오고 있어 모든 도전을 신속하게 해결하고 섬을 탈출해야 합니다.


<br>

### [문제 풀이 팁]

신비의 섬을 나가기 위해 필요한 비밀번호를 변수에 담아 놓는다면 번거로운 계산 과정이 줄어듭니다.


- 아직 변수에 대해 배운 적이 없다면?


__Chapter02 강의 예습 혹은 검색을 통해 간단한 개념을 파악할 수 있습니다!__


<br>

##### __<신비의 섬 탐험> 문제는 성장을 위해 난이도가 살짝 높게 작성되었기 때문에 어려울 수 있지만 도전해보시길 바랍니다 :)__



<br>

__[문제 1: 입구의 수수께끼]__

<br>

섬의 입구에 도착한 당신은 좋지 않은 느낌을 받습니다. 무언가 비릿한 냄새가 여기저기에서 납니다.

아마도 나보다 앞서 도착한 도전자들의 시체가 썩는 냄새인 것 같습니다.

지금...누군가 당신을 지켜보고 있습니다. 한시라도 빠르게 이 섬을 탈출해야 합니다.

우선 이 거대한 돌문을 열고 섬의 더 깊숙한 곳으로 들어가야 합니다. 돌문을 열려면 두 개의 열쇠 숫자를 더해야 합니다.

열쇠 숫자는 14와 29입니다.

돌문을 여는 숫자를 구해주세요.

In [68]:
key1 = 14
key2 = 29
pwd = key1 + key2
print(pwd)

43


<br>

__[문제 2: 첫 번째 강의 다리]__

<br>



돌문을 지나자 거친 강물이 흐르는 강이 나타났습니다.

아주 낡은 나무 다리가 힘겹게 메달려 있습니다.

당장이라도 무너질 것 같지만 이 다리 말고는 반대편으로 갈 수 있는 길이 보이지 않습니다.

무언가 불길한 기운이 계속 가까워지는 것만 같아 당신 빠르게 다리를 건너야 합니다.

다리는 50kg 이상 버티지 못하기 때문에 68kg인 당신은 무게를 줄여야 합니다.

방금 전 돌문을 통과하면서 얻은 마법 룬을 이용해 무게를 줄일 수 있습니다.

다리를 건너기 위해 줄여야 할 최소 무게를 구해주세요. (이상, 이하, 초과, 미만은 신경쓰지 않습니다!)


<br>

In [69]:
cur_weight = 68
limit = 50
print(cur_weight - limit, "kg 빼야 합니다")

18 kg 빼야 합니다




<br>

__[문제 3: 숲 속의 고목나무]__

<br>



첫 번째 강을 건너자 으스스한 숲 속에 들어섰습니다.

귀를 찢는 울음소리가 숲 속에 울려 퍼집니다. 당신은 바로 오래된 고목나무 숲으로 몸을 숨깁니다.

숲 깊이 들어가자 고목 나무에 걸린 현판이 보입니다.

"나무의 나이를 구하고 연못으로 가라"

나무의 나이를 구하기 위해서는 4년마다 한 번의 나이테가 생깁니다.

돌문을 열때 사용한 숫자를 나이테 개수로 사용하여 나무의 나이를 구하세요.



__Tips__
- 문제 1의 결과값을 참고하세요 !

In [70]:
tree_age = pwd * 4
print(tree_age)

172


<br>

__[문제 4: 숲 속의 마법 연못]__

<br>



고목나무를 지나자 신비로운 마법 연못이 나타났습니다.

연못 속에는 불가사의한 에너지가 흐르고 있습니다.

이 에너지를 이용하면 보물을 찾는 데 필요한 단서를 얻을 수 있습니다.

연못의 깊이를 측정하려면 나무의 나이를 이용해야 합니다.

나무의 나이가 172살이고, 연못의 깊이는 나무 나이의 1/4입니다.

연못의 깊이를 구하세요.

<br>



In [71]:
pond_age = tree_age // 4
print(pond_age)

43


<br>

__[문제 5: 사원의 입구]__

<br>



연못을 지나니 고대 사원이 나타났습니다. 사원의 입구에는 비밀번호를 입력해야 합니다.

비밀번호는 3자리 수로, 첫 번째 자리는 5, 두 번째 자리는 돌문을 여는 숫자의 십의 자리 수, 세 번째 자리는 돌문을 여는 숫자의 일의 자리 수의 3분의 1입니다.

비밀번호를 구하세요.

<br>

In [72]:
print("5", pwd // 10, pwd % 10 // 3)

5 4 1


<br>

__[문제 6: 비밀의 동굴]__

<br>

사원 내부에 비밀의 동굴로 이어지는 통로가 있습니다.

무언가 쿵 하고 넘어지는 소리가 들렸습니다. 신경을 곤두세우는 까마귀 소리가 여기저기서 울려 퍼집니다.

주변을 멤돌던 존재가 사원에 진입한것 같습니다.

어서 동굴을 벗어나야할 것 같습니다.

동굴을 열려면 암호를 풀어야 합니다. 암호는 돌문을 열때 사용한 숫자의 제곱근과 같습니다.

암호를 구하세요.

<br>

In [73]:
print(int(pwd ** 0.5))

6


<br>

__[문제 7: 해변의 파도]__

<br>


비밀의 동굴을 지나 해변에 도착했습니다.

동굴에서 빠져나오진 못한 존재는 엄청난 촉수를 가진 거대 외계인으로 보입니다. 외계인은 당신을 보며 군침을 흘리고 있습니다. 빠르게 이곳을 탈출해 섬의 반대편으로 도망가야 합니다.

배를 타고 빠르게 벗어나기 위해 우선 파도를 파악해야합니다.

파도가 일정한 간격으로 오고 있음을 발견했습니다.

5분 동안 20번의 파도가 왔다면, 파도 한 번이 오는 데 걸리는 평균 시간을 구하세요.

<br>

In [74]:
print(5 * 60 // 20, "secs")

15 secs


<br>

__[문제 8: 폭포의 높이]__

<br>

섬의 반대편의 무사히 도착했습니다.

거대한 폭포가 당신의 길을 가로 막고 있습니다.

자세히 보니 폭포 위쪽에 반짝이는 무언가가 있습니다. 당신이 찾던 보물일 수 있습니다.

마법 룬을 이용해 폭포의 높이를 구하고 폭포의 위로 올라갑니다.

물방울이 떨어져 땅에 닿는 데 걸리는 시간은 2초입니다. 중력가속도는 9.8m/s²입니다.

돌문을 열때 사용한 숫자를 이용해 폭포의 높이를 구하세요. (단, 공기저항은 무시합니다.)

<br>


__Tips__
- 높이 = 0.5 * 중력가속도 * (시간 + 돌문의 숫자의 마지막 자리 수)²

<br>

In [84]:
gravity_accel = 9.8
print(f'{0.5 * gravity_accel * ((2 + pwd % 10) ** 2):.2f}')

122.50


<br>

__[최종 문제: 탈출 비밀번호]__

<br>

보물을 찾았습니다.




쿠우에에이이이이이이


에일리언이 당신을 쫓아왔습니다. 당신은 곧 에일리언의 좋은 영양소가 될 것 같습니다.

당신은 지금 정신을 차리고 앞에 보이는 에일리언의 우주선을 뻇어 섬을 탈출해야합니다.

우주선에 타기 위해서는 비밀번호를 입력해야 합니다.

비밀번호는 현재 좌표의 첫 번째 숫자와 세 번째 숫자의 차와 같습니다.

비밀번호를 알아내세요!

<br>

현재 좌표 : 32, 43, 128

<br>

In [81]:
first_num = 32
third_num = 128
print(third_num - first_num)

96


<br>

당신은 생존했습니다.

생존한 당신은 보물 상자를 열어봅니다.



![image.png](https://t1.daumcdn.net/brunch/service/user/f2nO/image/5kAK9UfcMgaAGLdejuEuAoS0Hos.jpg)

-----
 [편지]

 오늘도 고생하셨습니다.

 내일도 과제 있습니다. 여러분 : )

 -파머-


---

__[문제1]__

<br>
변수명 게임에 들어 있는 값 중 '홀'만 출력되도록 코드를 작성해 주세요

game = "홀짝홀짝홀짝"

출력 예시 : "홀홀홀"


__Tips__
- 문자열 슬라이싱을 이용해보세요 !

In [ ]:
game = "홀짝홀짝홀짝"
print(game[0 : len(game) : 2])
# print(game[::2])

---

__[문제2]__

<br>

변수명 전화번호에 들어 있는 번호를 모두 붙여서 출력되도록 코드를 작성해주세요.

<br>

phone = "010-1234-5678"

출력 예시 : "01012345678"

__Tips__
- split() 함수는 문자열을 특정 구분자를 기준으로 나누어 리스트로 반환합니다.

In [ ]:
phone = "010-1234-5678"
print(''.join(phone.split('-')))

----

<br>

__[문제3]__

<br>

오즈 코딩 스쿨 백엔드 8기 과정에 참여한 수강생 중 5명의 수강생이 자율적으로 스터디 그룹을 결성했습니다.

스터디 그룹 내에는 한 주간 배운 내용으로 매주 쪽지 시험을 보는 룰이 있습니다.

이 쪽지 시험의 평균이 70점이 넘으면 주말 스터디는 없습니다.

주말 스터디의 여부를 True 또는 False로 출력해주세요.

<br>
A: 70, B: 75, C: 50, D: 80, E: 60

<br>

In [ ]:
A = 70
B = 75
C = 50
D = 80
E = 60

avg = (A + B + C + D + E) / 5
print(avg <= 70)

__[문제4]__

<br>

창의성, 시장성, 안정성, 수익성에 대한 각각의 평가 중 아래 기준에 부합하지 못하면 창업지원을 받을 수 없습니다.

- 수익성 60점 이상
- 창의성과 시장성 50점 이상
- 시장성과 수익성을 더한 값의 절반이 안정성 보다 높아야 함


다음 점수가 입력되었을때 창업 지원이 가능한지 여부를 True 또는 False로 출력해주세요.

---

창의성 : 62
시장성 : 71
안정성 : 68
수익성 : 66
<br>

* 참고
(creativity ,
marketability ,
stability ,
profitability )

__Tips__
* 비교 연산자와 논리 연산자의 결과를 고려해보아요

In [ ]:
creativity = 62
marketability = 71
stability = 68
profitability = 66

print(profitability >= 60 and creativity >= 50 and marketability >= 50 and ((marketability + profitability) / 2) > stability)

## 모험가들의 레벨업 도전

### [배경 이야기]

당신은 마법사, 전사, 도적, 힐러 네 명의 모험가를 이끄는 리더입니다.

각 캐릭터는 힘, 민첩, 지능, 체력이라는 능력치를 가지고 있습니다. 이 능력치들을 이용해 다양한 도전을 해결하고 사냥을 통해 경험치를 얻어 레벨업을 해야 합니다.

<br>

### [동료 캐릭터와 능력치 정보]

마법사: 힘(10), 민첩(20), 지능(80), 체력(50)

전사: 힘(80), 민첩(30), 지능(20), 체력(100)

도적: 힘(40), 민첩(80), 지능(30), 체력(60)

힐러: 힘(20), 민첩(40), 지능(70), 체력(70)

<br>

* 변수명을 영어로 변경해서 풀어보세요 !

<br>

__[문제 1: 첫 번째 사냥]__

마법사는 몬스터를 사냥하기 위해 마법을 사용합니다.

마법을 사용하려면 지능이 50 이상이어야 합니다.

마법사의 지능이 50 이상인지 확인하세요.

In [31]:
characters = {
    'magician' : {
        'strength' : 10,
        'dexterity' : 20,
        'intelligence' : 80,
        'health' : 50
    },
    'warrior' : {
        'strength' : 80,
        'dexterity' : 30,
        'intelligence' : 20,
        'health' : 100
    },
    'thief' : {
        'strength' : 40,
        'dexterity' : 80,
        'intelligence' : 30,
        'health' : 60
    },
    'healer' : {
        'strength' : 20,
        'dexterity' : 40,
        'intelligence' : 70,
        'health' : 70
    }
}

print(characters['magician']['intelligence'] >= 50)

True


<br>

__[문제 2: 두 번째 사냥]__

전사는 강력한 공격을 하기 위해 자신의 힘을 사용합니다.

전사의 힘이 60 이상이고 체력이 80 이상인지 확인하세요.

In [32]:
print(characters['warrior']['strength'] >= 60 and characters['warrior']['health'] >= 80)

True


<br>

__[문제 3: 세 번째 사냥]__

도적은 민첩함을 이용해 몬스터를 회피하고 공격합니다.

도적의 민첩이 70 이상이거나 힘이 50 이상인지 확인하세요.

In [33]:
characters['thief']['dexterity'] >= 70 or characters['thief']['strength'] >= 50

True

<br>

__[문제 4: 네 번째 사냥]__

힐러는 파티의 체력을 회복시키기 위해 마법을 사용합니다.

힐러의 지능이 60 이상이고, 체력이 50 이상인지 확인하세요.

In [34]:
characters['healer']['intelligence'] >= 60 and characters['healer']['health'] >= 50

True

<br>

__[문제 5: 레벨업]__

모든 캐릭터가 사냥을 성공적으로 마쳤습니다. 이제 경험치를 얻어 레벨업을 해야 합니다.


각 캐릭터의 경험치가 100 이상인지 print()를 이용해 확인해주세요.

<br>

경험치는 아래와 같습니다.

마법사: 120

전사: 150

도적: 110

힐러: 90



In [57]:
XPs = {
    'magician' : 120,
    'warrior' : 150,
    'thief' : 110,
    'healer' : 90
}

for i in XPs:  # == XPs.keys()
    print(f'{i}\'s been upgraded?   {XPs[i] >= 100}')

magician's been upgraded?   True
warrior's been upgraded?   True
thief's been upgraded?   True
healer's been upgraded?   False
